src/markdown/mod.rs

```diff
@@ -0,0 +1,204 @@
+mod block;
+mod span;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Block {
+    Paragraph(Vec<Span>),
+    Blockquote(Vec<Block>),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Span {
+    Emphasis(Vec<Span>),
+    Literal(char),
+    Strikethrough(Vec<Span>),
+    Strong(Vec<Span>),
+    Text(String),
+    Underline(Vec<Span>),
+}
+
+/// Parse a string and turn it into a tree of Markdown elements.
+pub fn parse(content: &str) -> Vec<Block> {
+    block::parse_blocks(content)
+}
+
+/// Turn a tree of Markdown blocks into a Minecraft formatted string.
+pub fn to_minecraft_format(blocks: &[Block]) -> String {
+    let mut ret = String::new();
+
+    for block in blocks {
+        let next = match block {
+            Block::Paragraph(ref elements) => format_paragraph(elements),
+            Block::Blockquote(ref elements) => format_blockquote(elements),
+        };
+
+        ret.push_str(&next);
+    }
+
+    ret
+}
+
+fn format_blockquote(elements: &[Block]) -> String {
+    format!("> {}", to_minecraft_format(elements))
+}
+
+fn format_paragraph(elements: &[Span]) -> String {
+    format_spans(elements, &mut vec![])
+}
+
+fn format_spans(elements: &[Span], mut open_tags: &mut Vec<String>) -> String {
```

Comment:
Add some documentation about what the whole `open_tags` thing is about and how it works. Can we also possibly make those strings constants?